just ignore tuple for now (patrys plz check)

                print "name:%s; F: %s (%s); N: %s; V: %s; R: %s\n" %(name, flags, type(flags), e, v, r)


19:12:55 glen[pts/4]@carme-pld tmp/p$ /home/users/glen/rpm/BUILD/createrepo-0.9.1/genpkgmetadata.py  .
name:rpmlib(PayloadIsLzma); F: LE; N: 0; V: 4.4.6; R: 1
name:cacti-spine; F: ('MISSINGOK', None); N: None; V: None; R: None

Traceback (most recent call last):
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/genpkgmetadata.py", line 290, in <module>
    main(sys.argv[1:])
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/genpkgmetadata.py", line 209, in main
    mdgen.doPkgMetadata()
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/createrepo/__init__.py", line 195, in doPkgMetadata
    self.writeMetadataDocs(packages, directory)
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/createrepo/__init__.py", line 268, in writeMetadataDocs
    self.primaryfile.write(po.do_primary_xml_dump(reldir, baseurl=self.conf.baseurl))
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/createrepo/yumbased.py", line 305, in do_primary_xml_dump
    msg += self._dump_format_items()
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/createrepo/yumbased.py", line 164, in _dump_format_items
    msg += self._dump_requires()
  File "/home/users/glen/rpm/BUILD/createrepo-0.9.1/createrepo/yumbased.py", line 271, in _dump_requires
    prcostring += ''' flags="%s"''' % flags
TypeError: not all arguments converted during string formatting

--- createrepo-0.9.1/createrepo/yumbased.py~	2008-02-29 19:16:38.000000000 +0200
+++ createrepo-0.9.1/createrepo/yumbased.py	2008-02-29 19:16:47.275839346 +0200
@@ -267,7 +267,10 @@
         for (name, flags, (e,v,r),pre) in mylist:
             prcostring = '''      <rpm:entry name="%s"''' % name
             if flags:
-                prcostring += ''' flags="%s"''' % flags
+                if type(flags) == str:
+                    prcostring += ''' flags="%s"''' % flags
+                else:
+                    prcostring += ''' flags=""'''
                 if e:
                     prcostring += ''' epoch="%s"''' % e
                 if v:
